fieldpath/set.go

```diff
@@ -94,6 +94,53 @@ func (s *Set) Difference(s2 *Set) *Set {
 	}
 }
 
+// RecursiveDifference returns a Set containing elements which:
+// * appear in s
+// * do not appear in s2
+//
+// Compared to a regular difference, this recursively removes
+// all children from s
+// that are either children or members in s2
+func (s *Set) RecursiveDifference(s2 *Set) *Set {
```

Comment:
I haven't looked at this code a lot, but it seems pretty long. Other methods of this object delegate a lot to Children and Members, is this possible here?

Reply:
I'll look at it.
I know the difference was done for members and children separately, but in this case both are needed to do the diff.
Feel free to correct me on that.

Reply:
I moved the code to SetNodeMap which reduced it a little as well.